lumicks/pylake/kymotracker/detail/trace_line_2d.py

```diff
@@ -236,7 +247,7 @@ def traverse_line(
     line = np.array(indices_bwd + indices_fwd[1:])
 
     if len(line) > 0:
-        return KymoLineData(time_idx=line[:, 1], coordinate_idx=line[:, 0])
+        return KymoLineData(time_idx=line[:, 1].tolist(), coordinate_idx=line[:, 0].tolist())
```

Comment:
Do you need to convert to a list here? Why not make `KymoLineData` store `np.ndarray` instead? It's more efficient and less to type.

Reply:
I had settled on lists since `KymoLineData` is often used for appending. It's not so relevant for that specific algorithm `track_lines`, but more so for `track_greedy` where single points are appended sequentially.

Anyways, measuring is better than guessing, so I just benchmarked the performance difference on some actual Kymos, and it's negligible, so I'll change it to numpy arrays (at least we'd have a more compact memory representation then).

numpy arrays, data classes w/no slots `31.83 +- 1.1`
lists, dataclass w/ slots `30.34 +- 1.9`

I kept the changes in a separate commit so it's easier to review.

Reply:
I think that for our purposes, `np.ndarray` is always the correct default for ints and floats. And then consider switching to something else if it turns out to be slow.